homeassistant/components/device_tracker/unifi.py

```diff
@@ -0,0 +1,100 @@
+""" Supports scanning a unifi router. """
+import logging
+from datetime import timedelta
+import threading
+
+from homeassistant.const import CONF_HOST, CONF_USERNAME, CONF_PASSWORD
+from homeassistant.helpers import validate_config
+from homeassistant.util import Throttle
+from homeassistant.components.device_tracker import DOMAIN
+
+# Return cached results if last scan was less then this time ago
+MIN_TIME_BETWEEN_SCANS = timedelta(seconds=5)
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_PORT = "port"
+CONF_VERSION = "version"
+CONF_SITEID = "siteid"
+
+
+def get_scanner(hass, config):
+    """ Validates config and returns a unifi scanner. """
+    if not validate_config(config,
+                           {DOMAIN: [CONF_HOST, CONF_USERNAME, CONF_PASSWORD, CONF_PORT, CONF_VERSION, CONF_SITEID]},
+                           _LOGGER):
+        return None
+
+    info = config[DOMAIN]
+
+    scanner = unifiDeviceScanner(
+        info[CONF_HOST], info[CONF_USERNAME], info[CONF_PASSWORD], info[CONF_PORT], info[CONF_VERSION], info[CONF_SITEID])
+
+    return scanner if scanner.success_init else None
+
+
+class unifiDeviceScanner(object):
+    """ This class queries a Unifi controller server. """
+
+    def __init__(self, host, username, password, port, version, siteid):
+        self.last_results = []
+
+        try:
+            # Pylint does not play nice if not every folders has an __init__.py
+            # pylint: disable=no-name-in-module, import-error
+            from unifi.controller import Controller
```

Comment:
Please add a requirement for this package in [requirements.txt](https://github.com/balloob/home-assistant/blob/master/requirements.txt)